core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -103,7 +104,12 @@ class RemoveSnapshots implements ExpireSnapshots {
 
   @Override
   public ExpireSnapshots cleanExpiredFiles(boolean clean) {
-    this.cleanExpiredFiles = clean;
+    LOG.warn("cleanExpiredFiles(boolean) is deprecated. Use cleanMode(CleanupMode) instead.");
```

Comment:
I don't think we have a pattern of logging warnings for API usages that are deprecated, so I don't think we should warn here